### PR TITLE
Resolve docker ENV PATH is not correctly interpreted in su commands in integTest

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Resolve docker ENV PATH is not correctly interpreted in su commands in integTest

### Issues Resolved
https://github.com/opensearch-project/opensearch-build-libraries/issues/197

```

Run command: su `id -un 1000` -c "env PATH=$PATH JAVA_HOME=/opt/java/openjdk-17 ./test.sh integ-test manifests/2.7.0/opensearch-2.7.0-test.yml --component geospatial --test-run-id <> --paths opensearch=/var/jenkins/workspace/bundle-build-zhujiaxi/deb "
[Pipeline] sh
+ id -un 1000
+ su opensearch -c env PATH=/usr/share/opensearch/.nvm/versions/node/v16.14.2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin JAVA_HOME=/opt/java/openjdk-17 ./test.sh integ-test manifests/2.7.0/opensearch-2.7.0-test.yml --component geospatial --test-run-id <> --paths opensearch=/var/jenkins/workspace/bundle-build-zhujiaxi/deb 
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tests/jenkins/TestRunIntegTestScript.groovy [hunk 1]
```diff
@@ -82,14 +82,14 @@ class TestRunIntegTestScript extends BuildPipelineTest {
     @Test
     void 'IntegTest LocalPath SwitchNonRoot=false'() {
         runScript("tests/jenkins/jobs/RunIntegTestScript_LocalPath_Jenkinsfile")
-        assertThat(getShellCommands('sh', 'test.sh'), hasItems('  ./test.sh integ-test tests/data/opensearch-1.3.0-test.yml --component OpenSearch --test-run-id null --paths opensearch=tests/jenkins/artifacts/tar '))
+        assertThat(getShellCommands('sh', 'test.sh'), hasItems('env  ./test.sh integ-test tests/data/opensearch-1.3.0-test.yml --component OpenSearch --test-run-id null --paths opensearch=tests/jenkins/artifacts/tar '))
```
Comment (gaiksaya) [inline tests/jenkins/TestRunIntegTestScript.groovy:85]: the env should have java_home right? Its going null here

Comment (peterzhuamazon) [reply tests/jenkins/TestRunIntegTestScript.groovy:85]: I have a fix now java_home is not always needed, but path can stay so will just add that.

Comment (peterzhuamazon) [reply tests/jenkins/TestRunIntegTestScript.groovy]: Added fix.

File: vars/runIntegTestScript.groovy [hunk 2]
```diff
@@ -38,13 +38,14 @@ void call(Map args = [:]) {
     }
     echo "Switch User to Non-Root (uid=1000): ${switchUser}"
 
-    String switchCommandStart = switchUser.equals('true') ? 'su `id -un 1000` -c "' : ''
+    // Avoid issue related to docker ENV keyword is not correctly interpreted in su commands
+    // https://github.com/opensearch-project/opensearch-build-libraries/issues/197
+    String switchCommandStart = switchUser.equals('true') ? "su `id -un 1000` -c \"env PATH=\$PATH $javaHomeCommand" : "env PATH=\$PATH $javaHomeCommand"
```
Comment (gaiksaya) [inline vars/runIntegTestScript.groovy:43]: Still not getting how PATH=$PATH matters if javaHomeCommand is added.
`env PATH=\$PATH JAVA_HOME=/opt/java/${javaVersion}` Should be 
`env PATH=\$PATH:/opt/java/${javaVersion}`

Comment (peterzhuamazon) [reply vars/runIntegTestScript.groovy:43]: No need, gradle is only looking for JAVA_HOME or java command in path.

Comment (peterzhuamazon) [reply vars/runIntegTestScript.groovy:43]: Even for the case of adding in path, it still needs to be running as ` env PATH=\$PATH:/opt/java/${javaVersion}/bin` not ` env PATH=\$PATH:/opt/java/${javaVersion}`

Comment (gaiksaya) [reply vars/runIntegTestScript.groovy:43]: I see! So we are adding JAVA_HOME as env and not adding java home path.

Comment (peterzhuamazon) [reply vars/runIntegTestScript.groovy:43]: yes.

Comment (gaiksaya) [reply vars/runIntegTestScript.groovy:43]: Can you add test case where java home is added? All above tests only add PATH?

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**